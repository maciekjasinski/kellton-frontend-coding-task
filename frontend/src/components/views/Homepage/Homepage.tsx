import { Link } from 'react-router-dom';

export const Homepage = () => {
  return (
    <div className="flex min-h-dvh flex-col items-center justify-center gap-8 px-4">
      <h1 className="text-2xl font-bold uppercase text-white">Lego minifigs mystery box</h1>
      <Link to="/choose-minifig" className="w-full">
        <button className="btn btn-primary w-full">Let's go!</button>
      </Link>
    </div>
  );
};
