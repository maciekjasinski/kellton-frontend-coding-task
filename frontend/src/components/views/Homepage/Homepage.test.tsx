import { describe, it, expect } from 'vitest';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { Homepage } from './Homepage';

describe('Homepage', () => {
  it('should create snapshot', () => {
    const page = renderer.create(
      <BrowserRouter>
        <Homepage />
      </BrowserRouter>,
    );
    expect(page).toMatchSnapshot();
  });

  it('page should have a title', () => {
    render(
      <BrowserRouter>
        <Homepage />
      </BrowserRouter>,
    );
    expect(screen.getByText('Lego minifigs mystery box')).toBeDefined();
  });
});
