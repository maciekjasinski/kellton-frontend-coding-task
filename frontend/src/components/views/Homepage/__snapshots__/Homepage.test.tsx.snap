// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage > should create snapshot 1`] = `
<div
  className="flex min-h-dvh flex-col items-center justify-center gap-12 px-4"
>
  <h1
    className="text-2xl font-extrabold uppercase text-white lg:text-4xl"
  >
    Lego minifigs mystery box
  </h1>
  <a
    className="w-full md:w-fit"
    href="/choose-minifig"
    onClick={[Function]}
  >
    <button
      className="shadow-buttonShadow btn btn-primary w-full md:w-fit"
    >
      Let's go
    </button>
  </a>
</div>
`;
